### PR TITLE
ClassCastExceptionForTopQueryWithAggregate

### PR DESCRIPTION
**Pull request by xinlian12**
https://github.com/Azure/azure-sdk-for-java/issues/12531

Refactored the code so we can skip the if else check in the TopDocumentQueryExecutionContext. Similar idea as how we pass the conitnuationToken along the pipeline.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/AggregateDocumentQueryExecutionContext.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
-import java.util.function.Function;
+import java.util.function.BiFunction;
```
Comment (kirankumarkolli) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/AggregateDocumentQueryExecutionContext.java:23]: Please update title with more user context: Like what scenario/state user sees.

Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/AggregateDocumentQueryExecutionContext.java:23]: updated to ClassCastExceptionForTopQueryWithAggregate~

**Pull request merged**